swupd/files.go

```diff
@@ -271,11 +271,18 @@ func (f *File) findFileNameInSlice(fs []*File) *File {
 	return nil
 }
 
-func (f *File) findIManifestInSlice(fs []*File) *File {
-	// get bundle name without IManifest version
-	prefix := strings.SplitAfter(f.Name, ".I.")[0]
-	for _, file := range fs {
-		if strings.HasPrefix(file.Name, prefix) {
+func (f *File) findBundleInManifest(manifest *Manifest) *File {
+	if f.Type != TypeManifest && f.Type != TypeIManifest {
+		return nil
+	}
+
+	name_split := func(str string) string {
```

Comment:
This is not idiomatic function naming in Go, lint is probably crying about it. I don't think you need it since you will only call once, otherwise update to `nameSplit()`.